android/src/com/unciv/app/MultiplayerTurnCheckWorker.kt

```diff
@@ -205,9 +224,33 @@ class MultiplayerTurnCheckWorker(appContext: Context, workerParams: WorkerParame
     override fun doWork(): Result {
         val showPersistNotific = inputData.getBoolean(PERSISTENT_NOTIFICATION_ENABLED, true)
         val configuredDelay = inputData.getInt(CONFIGURED_DELAY, 5)
+
         try {
-            val currentTurnPlayer = OnlineMultiplayer().tryDownloadCurrentTurnCiv(inputData.getString(GAME_ID)!!)
-            if (currentTurnPlayer.playerId == inputData.getString(USER_ID)!!) {
+            val gameIds = inputData.getStringArray(GAME_ID)!!
+            val gameNames = inputData.getStringArray(GAME_NAMES)!!
+            var arrayIndex = 0
+            // We only want to notify the user or update persisted notification once but still want
+            // to download all games to update the files hence this bool
+            var foundGame = false
+
+            for (gameId in gameIds){
+                //gameId could be an empty string if startTurnChecker fails to load all files
+                if (gameId.isEmpty())
+                    continue
+
+                val game = OnlineMultiplayer().tryDownloadGameUninitialized(gameId)
+                val currentTurnPlayer = game.getCivilization(game.currentPlayer)
+
+                //Save game so MultiplayerScreen gets updated
+                GameSaver.saveGame(game, gameNames[arrayIndex], true)
```

Comment:
I think this should get its own try-catch block because saving the game is not essential for the TurnChecker.
However, this would increase the indentation.